clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -687,7 +702,14 @@ protected boolean handleControlMessage(
           pubSubTopicPartition.getPartitionNumber());
     }
     if (controlMessageType.equals(ControlMessageType.VERSION_SWAP)) {
-      return handleVersionSwapControlMessage(controlMessage, pubSubTopicPartition, topicSuffix);
+      // TODO: In view topics, we need to know the partition of the upstream RT
+      // how we transmit this information has yet to be determined, so once we finalize
+      // that, we'll need to tweak this. For now, we'll just pass in the same partition number
+      return handleVersionSwapControlMessage(
+          controlMessage,
+          pubSubTopicPartition,
+          topicSuffix,
+          pubSubTopicPartition.getPartitionNumber());
```

Comment:
Could you elaborate a bit how will this work in this PR? i.e. without the proper upstream RT partition. Since we don't know the mapping shouldn't we set the `currentVersionHighWatermarks` with the lowest watermark from version swap? Otherwise wouldn't we filter out messages unintentionally?

Reply:
Unless we know the upstream RT partition no offset highwatermark is actually comparable.  So I don't think keeping the 'lowest' is the right answer.  Really without knowing the source RT partition all answers are wrong